libraries/botbuilder-dialogs-adaptive/src/dynamicBeginDialogDeserializer.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Dialog } from 'botbuilder-dialogs';
+import { CustomDeserializer, ResourceExplorer } from 'botbuilder-dialogs-declarative';
+import { BeginDialogConfiguration, DynamicBeginDialog } from './actions';
+
+/**
+ * Internal serializer for `DynamicBeginDialog` which bind the x.dialog resourceId to the DynamicBeginDialog.dialog property.
+ */
+export class DynamicBeginDialogDeserializer
+    implements CustomDeserializer<DynamicBeginDialog, BeginDialogConfiguration> {
+    /**
+     * Intializes an instance of `DynamicBeginDialogDeserializer`.
+     * @param _resourceExplorer The `ResourceExplorer` used by the deserializer.
+     * @param _resourceId The resource id of the dynamic dialog.
+     */
+    public constructor(private readonly _resourceExplorer: ResourceExplorer, private readonly _resourceId: string) {}
+
+    /**
+     * The method that loads the configuration object to a `DynamicBeginDialog` object.
+     * @param config The configuration object to deserialize.
+     * @param type The object type that the configuration will be deserialized to.
+     * @returns A `DynamicBeginDialog` object created from the configuration.
+     */
+    public load(
+        config: BeginDialogConfiguration,
+        type: { new (...args: unknown[]): DynamicBeginDialog }
+    ): DynamicBeginDialog {
+        config.dialog = this._resourceExplorer.loadType<Dialog>(this._resourceId);
+        return Object.entries(config).reduce((instance, [key, value]) => {
+            let converter = instance.getConverter(key as keyof BeginDialogConfiguration);
```

Comment:
> Does this include a fix for #2234? It looked like there was a draft PR linked that might be superseded by this PR.
> 
> EDIT: Looks like #2142 was also linked to the draft PR.

No, I would prefer fixing these two issues in another PR after this.

Reply:
For a `DynamicBeginDialog`, there might be some user defined properties, we cannot limit the key of object to `BeginDialogConfiguration`, so I have to use `as keyof` to avoid build errors.